include/zephyr/net/net_config.h

```diff
@@ -26,6 +27,131 @@ extern "C" {
  * @{
  */
 
+/** @cond INTERNAL_HIDDEN */
+
+#if defined(CONFIG_NET_CONFIG_USE_YAML)
+#include "net_init_config.inc"
+#else
+#if !defined(NET_CONFIG_IFACE_COUNT)
+#define NET_CONFIG_IFACE_COUNT 0
+#endif
+#if !defined(NET_CONFIG_DHCPV4_SERVER_INSTANCE_COUNT)
+#define NET_CONFIG_DHCPV4_SERVER_INSTANCE_COUNT 0
+#endif
+#endif
+
+struct net_init_config_ipv6_prefix {
+	struct in6_addr address;
+	uint32_t lifetime;
+	uint8_t len;
+};
+
+struct net_init_config_dhcpv6 {
+	bool do_request_addr : 1;
+	bool do_request_prefix : 1;
+	bool is_enabled : 1;
+};
+
+/* Compliance checker complains about this valid macro so use a define */
+#define __z_net_config_check_dhcpv6 \
```

Comment:
Is this approach really adding to readability? Why not the less indirect approach of #if / #endif directly in the struct? What am I overlooking here?

Reply:
I needed to add some trickery because our compliance checker script started to complain how things were implemented wrongly. That is why the current code looks like this. There are probably other ways to make compliance pass, I can investigate this in the future.